registry/txt.go

```diff
@@ -201,26 +206,37 @@ type nameMapper interface {
 	toTXTName(string) string
 }
 
-type prefixNameMapper struct {
+type affixNameMapper struct {
 	prefix string
+	suffix string
 }
 
-var _ nameMapper = prefixNameMapper{}
+var _ nameMapper = affixNameMapper{}
 
-func newPrefixNameMapper(prefix string) prefixNameMapper {
-	return prefixNameMapper{prefix: strings.ToLower(prefix)}
+func newaffixNameMapper(prefix string, suffix string) affixNameMapper {
+	return affixNameMapper{prefix: strings.ToLower(prefix), suffix: strings.ToLower(suffix)}
 }
 
-func (pr prefixNameMapper) toEndpointName(txtDNSName string) string {
+func (pr affixNameMapper) toEndpointName(txtDNSName string) string {
 	lowerDNSName := strings.ToLower(txtDNSName)
-	if strings.HasPrefix(lowerDNSName, pr.prefix) {
+	if strings.HasPrefix(lowerDNSName, pr.prefix) && len(pr.suffix) == 0 {
 		return strings.TrimPrefix(lowerDNSName, pr.prefix)
 	}
+
+	if len(pr.suffix) > 0 {
+		regex := regexp.MustCompile(`\.`)
+		DNSName := regex.Split(lowerDNSName, 2)
+		if strings.HasSuffix(DNSName[0], pr.suffix) {
+			return strings.TrimSuffix(DNSName[0], pr.suffix) + "." + DNSName[1]
+		}
+	}
 	return ""
 }
 
-func (pr prefixNameMapper) toTXTName(endpointDNSName string) string {
-	return pr.prefix + endpointDNSName
+func (pr affixNameMapper) toTXTName(endpointDNSName string) string {
+	regex := regexp.MustCompile(`\.`)
+	DNSName := regex.Split(endpointDNSName, 2)
```

Comment:
Same as above, can we use `strings.Split` ?

Reply:
Done, thanks for the hint